.github/workflows/ci_tests.yml

```diff
@@ -0,0 +1,218 @@
+            DOCKER_REPOSITORY: geosx/pangea2-gcc8.3.0-openmpi2.1.5-mkl2019.3
+            CMAKE_BUILD_TYPE: Release
+            GCP_BUCKET: geosx/Pangea2
+            ENABLE_HYPRE: ON
+            ENABLE_TRILINOS: OFF
+
+          - name: Sherlock CPU (centos 7.9.2009, gcc 10.1.0, open-mpi 4.1.2, openblas 0.3.10)
+            DOCKER_REPOSITORY: geosx/sherlock-gcc10.1.0-openmpi4.1.2-openblas0.3.10-zlib1.2.11
+            CMAKE_BUILD_TYPE: Release
+            HOST_CONFIG: host-configs/Stanford/sherlock-gcc10-ompi4.1.2-openblas0.3.10.cmake
+            GCP_BUCKET: geosx/Sherlock-CPU
+            ENABLE_HYPRE: ON
+            ENABLE_TRILINOS: OFF
+
+          - name: Ubuntu (22.04, gcc 11.2.0, open-mpi 4.1.2)
+            DOCKER_REPOSITORY: geosx/ubuntu22.04-gcc11
+            CMAKE_BUILD_TYPE: Release
+            GCP_BUCKET: geosx/ubuntu22.04-gcc11
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+        lfs: false
+
+    - id: 'auth'
+      if: matrix.GCP_BUCKET
+      uses: 'google-github-actions/auth@v1'
+      with:
+        credentials_json: '${{ secrets.GOOGLE_CLOUD_GCP }}'
+    - name: 'Set up Cloud SDK'
+      if: matrix.GCP_BUCKET
+      uses: 'google-github-actions/setup-gcloud@v1'
+      with:
+        version: '>= 363.0.0'
+
+    - name: Print environment
+      run: printenv
+
+    # Build and test only
+    # Builds only the geosx executable (timeout when building tests)
+    - name: Build and test
+      if: ${{ !(matrix.GCP_BUCKET) }}
+      env:
+        DOCKER_REPOSITORY: ${{ matrix.DOCKER_REPOSITORY }}
+        CMAKE_BUILD_TYPE: ${{ matrix.CMAKE_BUILD_TYPE }}
+        BUILD_AND_TEST_ARGS: ${{ matrix.BUILD_AND_TEST_ARGS }}
+        ENABLE_HYPRE: ${{ matrix.ENABLE_HYPRE }}
+        ENABLE_HYPRE_DEVICE: ${{ matrix.ENABLE_HYPRE_DEVICE }}
+        ENABLE_TRILINOS: ${{ matrix.ENABLE_TRILINOS }}
+      run: ./scripts/ci_build_and_test.sh
+
+    # Build, test, uploads GEOSX and its TPL to GCP/GCS using gcloud CLI
+    - name: Build and test and deploy
+      if: matrix.GCP_BUCKET
+      env:
+        DOCKER_REPOSITORY: ${{ matrix.DOCKER_REPOSITORY }}
+        CMAKE_BUILD_TYPE: ${{ matrix.CMAKE_BUILD_TYPE }}
+        BUILD_AND_TEST_ARGS: ${{ matrix.BUILD_AND_TEST_ARGS }}
+        HOST_CONFIG: ${{ matrix.HOST_CONFIG }}
+        ENABLE_HYPRE: ${{ matrix.ENABLE_HYPRE }}
+        ENABLE_TRILINOS: ${{ matrix.ENABLE_TRILINOS }}
+        GCP_BUCKET: ${{ matrix.GCP_BUCKET }}
+      run: |
+        source ./scripts/ci_build_and_test.sh
+        TMP_DIR=/tmp
+        GEOSX_EXPORT_DIR=GEOSX-and-TPL-${GITHUB_SHA:0:7}
+        docker cp -a ${CONTAINER_NAME}:${GEOSX_TPL_DIR}/.. ${TMP_DIR}/${GEOSX_EXPORT_DIR}
+        GEOSX_BUNDLE=${TMP_DIR}/${GEOSX_EXPORT_DIR}.tar.gz
+        tar czf ${GEOSX_BUNDLE} --directory=${TMP_DIR} ${GEOSX_EXPORT_DIR}
+        CLOUDSDK_PYTHON=python3 gsutil cp -a public-read ${GEOSX_BUNDLE} gs://${GCP_BUCKET}/
```

Comment:
I think that eventually we should improve the `ci_build_and_test.sh` script such that it conveniently receives CLI arguments, removing the usage of implicit `env` variables.